Nvidia GPU with CUDA support  `cargo run` fails cmake with "CUDA not found"

**Describe the bug**

I  have all deps installed and pass tests, but try:

https://github.com/TabbyML/tabby/blob/b71f9c5c32bf3e28ce5fc0f5b49cf9d8ec3fa167/CONTRIBUTING.md?plain=1#L58-L62

<details>

<summary>Here is the error 🖯 </summary>

```
Caused by:
  process didn't exit successfully: `tabby/target/release/build/llama-cpp-bindings-a6a735ca3805874f/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=include/engine.h
  cargo:rerun-if-changed=src/engine.cc
  cargo:rustc-link-lib=llama
  cargo:rustc-link-lib=ggml_static
  cargo:rustc-link-search=native=/usr/local/cuda/lib64
  cargo:rustc-link-lib=culibos
  cargo:rustc-link-lib=cuda
  cargo:rustc-link-lib=cudart
  cargo:rustc-link-lib=cublas
  cargo:rustc-link-lib=cublasLt
  CMAKE_TOOLCHAIN_FILE_x86_64-unknown-linux-gnu = None
  CMAKE_TOOLCHAIN_FILE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-unknown-linux-gnu = None
  CMAKE_GENERATOR_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-unknown-linux-gnu = None
  CMAKE_PREFIX_PATH_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-unknown-linux-gnu = None
  CMAKE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "tabby/target/release/build/llama-cpp-bindings-a58174a4e6907958/out/build" && CMAKE_PREFIX_PATH="" "cmake" "tabby/crates/llama-cpp-bindings/llama.cpp" "-DLLAMA_NATIVE=OFF" "-DINS_ENB=ON" "-DCMAKE_BUILD_TYPE=Release" "-DLLAMA_CUBLAS=ON" "-DCMAKE_POSITION_INDEPENDENT_CODE=ON" "-DCMAKE_INSTALL_PREFIX=tabby/target/release/build/llama-cpp-bindings-a58174a4e6907958/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/bin/cc"
  -- The C compiler identification is GNU 11.4.0
  -- The CXX compiler identification is GNU 11.4.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /usr/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Found Git: /usr/bin/git (found version "2.34.1")
  -- Looking for pthread.h
  -- Looking for pthread.h - found
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
  -- Found Threads: TRUE
  -- Could not find nvcc, please set CUDAToolkit_ROOT.
  -- ccache found, compilation results will be cached. Disable with LLAMA_CCACHE=OFF.
  -- CMAKE_SYSTEM_PROCESSOR: x86_64
  -- x86 detected
  -- Configuring incomplete, errors occurred!
  See also "tabby/target/release/build/llama-cpp-bindings-a58174a4e6907958/out/build/CMakeFiles/CMakeOutput.log".

  --- stderr
  CMake Warning at CMakeLists.txt:391 (message):
    LLAMA_CUBLAS is deprecated and will be removed in the future.

    Use LLAMA_CUDA instead


  CMake Warning at CMakeLists.txt:458 (message):
    CUDA not found


  -- CUDA host compiler is GNU
  CMake Error at CMakeLists.txt:903 (get_flags):
    get_flags Function invoked with incorrect arguments for function named:
    get_flags


  thread 'main' panicked at /home/nuke/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5:

  command did not execute successfully, got: exit status: 1

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```
</details>

**Information about your version**

Using main @ [b71f9c5](https://github.com/TabbyML/tabby/commit/b71f9c5c32bf3e28ce5fc0f5b49cf9d8ec3fa167)


**Information about your GPU**

```
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.67                 Driver Version: 550.67         CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3060 ...    Off |   00000000:01:00.0 Off |                  N/A |
| N/A   47C    P8             11W /   40W |       9MiB /   6144MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A      3243      G   /usr/lib/xorg/Xorg                              4MiB |
+-----------------------------------------------------------------------------------------+
```


**Additional context**

Using [Pop!_OS](https://pop.system76.com/) 22.04 LTS with nvidia drivers installed, of course :grin:

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -45,6 +45,11 @@ This will help ensure everything is working correctly and avoid surprises with l
 
 Golden tests, which run models and check their outputs against previous "golden snapshots", should be skipped for most development purposes, as they take a very long time to run (especially the tests running the models on CPU). You may still want to run them if your changes relate to the functioning of or integration with the generative models, but skipping them is recommended otherwise.
 
+Optionally, to use a GPU make sure you have the correct drivers and libraries installed for your device:
+
+> **CUDA for Nvidia** - [Linux](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html), [Windows](https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html)
+> **ROCm for AMD** - [Linux](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/tutorial/quick-start.html), [Windows](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/)
+
 ## Building and Running
 
 Tabby can be run through `cargo` in much the same manner as docker:
@@ -53,7 +58,7 @@ Tabby can be run through `cargo` in much the same manner as docker:
 cargo run serve --model TabbyML/StarCoder-1B
 ```
 
-This will run Tabby locally on CPU, which is not optimal for performance. Depending on your GPU and its compatibility, you may be able to run Tabby with GPU acceleration. Please make sure you have CUDA or ROCm installed, for Nvidia or AMD graphics cards respectively. No extra library installation is necessary for Apple silicon (M1/M2) using Metal.
+This will run Tabby locally on CPU, which is not optimal for performance. Depending on your GPU and its compatibilities, you may be able to run Tabby with GPU acceleration. First insure you have the dependencies for your Nvidia or AMD GPU installed. No extra library installation is necessary for Apple silicon (M1/M2) using Metal.
 
 To run Tabby locally with CUDA (NVIDIA):
 

